src/socket.c

```diff
@@ -714,11 +714,15 @@ LIBIMOBILEDEVICE_GLUE_API int get_primary_mac_address(unsigned char mac_addr_buf
 			if (ifa->ifa_flags & IFF_LOOPBACK) {
 				continue;
 			}
-#if defined(__APPLE__)
+#if defined(__APPLE__) || defined (__FreeBSD__)
 			if (ifa->ifa_addr->sa_family != AF_LINK) {
 				continue;
 			}
+#if defined (__APPLE__)
 			if (!strcmp(ifa->ifa_name, "en0")) {
+#elif defined (__FreeBSD__)
+			{
```

Comment:
Is it guaranteed to be a regular interface and not for example the loopback device?

Reply:
There is a line just above this to check `ifa_flags` for whether is a loopback device.

Reply:
LOL and the best thing is I wrote it 🤣